### DISCUSSION TITLE
Webserver behind an Nginx Proxy Manager

### DISCUSSION THREAD

**Discussion by Storkiii** [aaronwmorris/indi-allsky#1662] [General]
I’m having trouble setting up a web-only server with my Nginx Proxy Manager (NPM).

The web server runs on an Ubuntu VM, the NPM is running as a Docker container and both are hosted on the same Unraid machine.
 I’m using a Cloudflare DNS, which directs all traffic to the NPM.

The issue is that any attempts to redirect traffic from NPM to the Ubuntu VM fail. I consistently get the following error:

SSL_ERROR_UNRECOGNIZED_NAME_ALERT

Here’s what I’ve tried so far:

Adding the domain as server name to the apache2 configuration file.
Generating a new SSL certificate using Certbot (but this fails every time).
Requesting a certificate directly through NPM (this also fails).

The NPM is handling redirects and certificates for other web applications without any issues.
But that are all "ready to go" Docker Containers with minimal configuration, that i could handle. 
But now iam out of ideas and hoping someone can help me.

**Comment by aaronwmorris** [comment]
My guess is the hostname in the URL you are using does not match the Common Name [CN] or Subject Alternative Name [SAN] fields of the certificate on the Apache side.

It sounds like NPM is trying to validate those fields and is failing.

**Comment by Storkiii** [comment]
I got it working. 
It would have saved me hours of headache if I had remembered earlier that I configured a DNS challenge for my domain.
But thanks for the fast response!

An off-topic question, if allowed:
Is it possible to offload the timelapse/keogram creation to the web server?

**Reply by aaronwmorris** [reply]
The question is fine... unfortunately, it is not possible.  It is difficult to operate long running processes with a web server because you generally have a maximum execution time of 180s for a process.  I know there are some ways of working around that, but the capture process is already has a dedicated process for long running tasks.

**Reply by Storkiii** [reply]
Okay. Thanks for the Info.
